robottelo/utils/sso.py

```diff
@@ -0,0 +1,197 @@
+        if not username:
+            username = gen_string('alphanumeric')
+        RHSSO_NEW_USER['username'] = username
+        RHSSO_NEW_USER['email'] = random.choice(valid_emails_list())
+        RHSSO_RESET_PASSWORD['value'] = settings.rhsso.rhsso_password
+        self.upload_rhsso_entity(RHSSO_NEW_USER, "create_user")
+        self.upload_rhsso_entity(RHSSO_RESET_PASSWORD, "reset_password")
+        self.execute(f"{KEY_CLOAK_CLI} create users -r {settings.rhsso.realm} -f create_user")
+        user_details = self.get_rhsso_user_details(RHSSO_NEW_USER['username'])
+        self.execute(
+            "{} update -r {} users/{}/reset-password -f {}".format(
+                KEY_CLOAK_CLI, settings.rhsso.realm, user_details['id'], "reset_password"
+            )
+        )
+        return RHSSO_NEW_USER
+
+    def update_rhsso_user(self, username, group_name=None):
+        user_details = self.get_rhsso_user_details(username)
+        RHSSO_USER_UPDATE['realm'] = f"{settings.rhsso.realm}"
+        RHSSO_USER_UPDATE['userId'] = f"{user_details['id']}"
+        if group_name:
+            group_details = self.get_rhsso_groups_details(group_name=group_name)
+            RHSSO_USER_UPDATE['groupId'] = f"{group_details['id']}"
+            self.upload_rhsso_entity(RHSSO_USER_UPDATE, "update_user")
+            group_path = f"users/{user_details['id']}/groups/{group_details['id']}"
+            self.execute(
+                f"{KEY_CLOAK_CLI} update -r {settings.rhsso.realm} {group_path} -f update_user"
+            )
+
+    def delete_rhsso_user(self, username):
+        """Delete the RHSSO user"""
+        user_details = self.get_rhsso_user_details(username)
+        self.execute(f"{KEY_CLOAK_CLI} delete -r {settings.rhsso.realm} users/{user_details['id']}")
+
+    def create_group(self, group_name=None):
+        """Create the RHSSO group"""
+        if not group_name:
+            group_name = gen_string('alphanumeric')
+        RHSSO_NEW_GROUP['name'] = group_name
+        self.upload_rhsso_entity(RHSSO_NEW_GROUP, "create_group")
+        result = self.execute(
+            f"{KEY_CLOAK_CLI} create groups -r {settings.rhsso.realm} -f create_group"
+        )
+        return result
+
+    def delete_rhsso_group(self, group_name):
+        """Delete the RHSSO group"""
+        group_details = self.get_rhsso_groups_details(group_name)
+        self.execute(
+            f"{KEY_CLOAK_CLI} delete -r {settings.rhsso.realm} groups/{group_details['id']}"
+        )
+
+    def update_client_configuration(self, json_content):
+        """Update the client configuration"""
+        client_id = self.get_rhsso_client_id()
+        self.upload_rhsso_entity(json_content, "update_client_info")
+        update_cmd = (
+            f"{KEY_CLOAK_CLI} update clients/{client_id}"
+            f"-f update_client_info -s enabled=true --merge"
+        )
+        self.execute(update_cmd)
+
+    def get_oidc_token_endpoint(self):
```

Comment:
we can use @property all of those getter methods and with the use of self.hostname